src/Visualization/Python/GenerateXdmf.py

```diff
@@ -312,11 +312,12 @@ def data_item(data_type):
 
 
 @click.command(help=generate_xdmf.__doc__)
-@click.option(
-    '--file-prefix',
-    required=True,
-    help=("The common prefix of the H5 volume files to load, excluding "
-          "the node number integer(s)"))
+@click.argument('h5files',
```

Comment:
It looks like if no h5 files are specified, nothing will happen (line 24 above). If a filename is specified but that file doesn't exist, it looks like you'll get an `h5py` error about not being able to open a file.

Reply:
The [click docs](https://click.palletsprojects.com/en/8.1.x/parameters/) agree that most things should be options (flags), except for input files / URLs. Arbitrary number of input files or globs only work with (positional) arguments. The documentation for them is part of the main help string. Here's the help message (edited a bit in a fixup to make the argument name clearer):

```
Usage: spectre generate-xdmf [OPTIONS] [H5FILES]...

  Generate an XDMF file for ParaView and VisIt

  Read volume data from the 'H5FILES' and generate an XDMF file. The XDMF file
  points into the 'H5FILES' files so ParaView and VisIt can load the volume
  data. To process multiple files suffixed with the node number, specify a
  glob like 'VolumeData*.h5'.

  To load the XDMF file in ParaView you must choose the 'Xdmf Reader', NOT
  'Xdmf3 Reader'.

Options:
  -o, --output PATH        Output file name, an xmf extension will be added
  -d, --subfile-name TEXT  Name of the volume data subfile in the H5 files. A
                           '.vol' extension is added if needed. If
                           unspecified, list all '.vol' subfiles and exit.
  --stride INTEGER         View only every stride'th time step
  --start-time FLOAT       The earliest time at which to start visualizing.
                           The start-time value is included.
  --stop-time FLOAT        The time at which to stop visualizing. The stop-
                           time value is not included.
  --coordinates TEXT       The coordinates to use for visualization  [default:
                           InertialCoordinates]
  -h, --help               Show this message and exit.
```

Reply:
@knelli2 if you specified a file that doesn't exist then you'll get a usage error before the code even gets to `h5py`, because of the `click.Path` checks in the argument`. Here's the error:

```
$ ./bin/spectre generate-xdmf file
Usage: spectre generate-xdmf [OPTIONS] [H5FILES]...
Try 'spectre generate-xdmf -h' for help.

Error: Invalid value for '[H5FILES]...': File 'file' does not exist.
```